src/mappings/spacewalk.ts

```diff
@@ -0,0 +1,188 @@
+import { network } from '../config'
+import { IssueRequestStatus } from '../model/generated/_issueRequestStatus'
+import { RedeemRequestStatus } from '../model/generated/_redeemRequestStatus'
+import { IssueRequestType, RedeemRequestType } from '../types/common'
+import { beautifyCurrencyIdString } from './token'
+import { deriveStellarPublicKeyFromHex } from './token'
+
+function getIssueRequestStatus(status: IssueRequestType['status']) {
+    switch (status.__kind) {
+        case 'Cancelled':
+            return IssueRequestStatus.CANCELLED
+        case 'Completed':
+            return IssueRequestStatus.COMPLETED
+        case 'Pending':
+            return IssueRequestStatus.PENDING
+    }
+}
+
+function getRedeemRequestStatus(status: RedeemRequestType['status']) {
+    switch (status.__kind) {
+        case 'Pending':
+            return RedeemRequestStatus.PENDING
+        case 'Completed':
+            return RedeemRequestStatus.COMPLETED
+        case 'Retried':
+            return RedeemRequestStatus.RETRIED
+        case 'Reimbursed':
+            if (status.value) {
+                return RedeemRequestStatus.REIMBURSED_MINTED
+            } else {
+                return RedeemRequestStatus.REIMBURSED
+            }
+    }
+}
+
+// This function is used to get the latest info about an issue request from the storage and save it to the database.
+async function createOrUpdateIssueRequest(
+    ctx: EventHandlerContext,
+    issueId: string
+) {
+    const issueRequestStorage = await getVersionedStorage(
+        network,
+        ctx,
+        'issue',
+        'issueRequests'
+    )
+    const storageIssue = (await issueRequestStorage.get(
+        ctx.block,
+        issueId
+    )) as IssueRequestType
+
+    const vault = await getOrCreateVault(
+        ctx,
+        storageIssue.vault,
+        storageIssue.stellarAddress
+    )
+
+    // Get existing issue request from the database if any
+    let existingIssueRequest = await ctx.store.get(IssueRequest, issueId)
+
+    const issueRequest = new IssueRequest({
+        id: issueId,
+        timestamp: existingIssueRequest
+            ? existingIssueRequest.timestamp
+            : new Date(ctx.block.timestamp ?? 0),
+        opentime: BigInt(storageIssue.opentime),
+        period: BigInt(storageIssue.period),
+        requester: maybeHexToSs58(storageIssue.requester),
+        amount: BigInt(storageIssue.amount),
+        vault: vault, // Our vault entity, not the storage's vault type
+        fee: BigInt(storageIssue.fee),
+        asset: beautifyCurrencyIdString(storageIssue.asset),
+        stellarAddress: deriveStellarPublicKeyFromHex(
+            storageIssue.stellarAddress
+        ),
+        griefingCollateral: BigInt(storageIssue.griefingCollateral),
+        status: getIssueRequestStatus(storageIssue.status),
+    })
+
+    await ctx.store.save(issueRequest)
+}
+
+export async function handleIssueRequest(ctx: EventHandlerContext) {
```

Comment:
Nit: Let's rename this and `handleRedeemRequest` to `handleXXRequestCreated`.

Reply:
You mean have only one for any request creation? But then we have to switch based on the event and is more or less the same right?

Reply:
Ah... You mean only add `....Created`. Sure.